src/rust/lib/system/web/src/lib.rs

```diff
@@ -481,3 +483,79 @@ pub mod traits {
     pub use super::NodeInserter;
     pub use super::NodeRemover;
 }
+
+/// Sleeps for the specified amount of time.
+///
+/// This function might sleep for slightly longer than the specified duration but never less.
+///
+/// This function is an async version of std::thread::sleep.
+#[cfg(target_arch = "wasm32")]
+pub async fn sleep(duration:Duration) {
+    use wasm_bindgen_futures::JsFuture;
+
+    let performance = performance();
+    let call_time   = performance.now();
+    let future : JsFuture = js_sys::Promise::new(&mut |resolve:Function,_| {
+        let time_elapsed_from_call = ((performance.now() - call_time) * 1000.0) as i32;
```

Comment:
This is interesting. Is it really measurable? I was expecting it to be really close to 0. What are the results here? I think it is worth mentioning in the comment inside the code (not even doc comment) what was the offset measured as people inspecting this code would constantly wonder about it.

Reply:
Well, yes... If we use `sleep(duration).await`, the time elapsed is close to 0. But I think there might be cases we wan't start a request, instantiate the asynchronous sleep function as a timeout, make other operations and then await it.

This is required in the original task decription: https://github.com/luna/ide/issues/351

Reply:
Oh, ok. @mwu-tow just responded at the same time :)